sshd-core/src/main/java/org/apache/sshd/server/forward/TcpipServerChannel.java

```diff
@@ -260,6 +260,20 @@ protected void handleChannelOpenFailure(OpenFuture f, Throwable problem) {
 
     @Override
     public CloseFuture close(boolean immediately) {
+
+        if (!immediately && (out != null)) {
+            try {
+                if (log.isDebugEnabled()) {
+                    log.debug("Closing ChannelOutputStream: {}", TcpipServerChannel.this);
+                }
+                out.close();
```

Comment:
1. Style  - parentheses around `(!immediately)`
2. Code seems a bit complex and not well documented to explain why this is done - why is closing what seems to be an `OutputStream` same as sending `SSH_MSG_CHANNEL_EOF`
3. If indeed `SSH_MSG_CHANNEL_EOF`  is required only in a certain scenario why not send it explicitly ? We do have a `sendOf()` method that can be invoked by the channel on itself (and it is even idempotent...)

Reply:
1. Ok, I'll change it
2. 3. You're right It's a little bit wired and I'll try to add some comments to explain it better. Anyway, after having reviewed the code, to me it seems that out.close() is more approriate than simply sendEof() because close() will also call flush(), avoiding some data not beign sent back to client.